setup.py

```diff
@@ -49,61 +53,24 @@
 
 
 setup(
-    name='JPype1',
-    version='1.5.0_dev0',
-    description='A Python to Java bridge.',
-    long_description=open('README.rst').read(),
-    license='License :: OSI Approved :: Apache Software License',
-    author='Steve Menard',
-    author_email='devilwolf@users.sourceforge.net',
-    maintainer='Luis Nell',
-    maintainer_email='cooperate@originell.org',
-    python_requires=">=3.7",
-    url='https://github.com/jpype-project/jpype',
+    # Non-standard, and extension behaviour. See also:
+    # https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration
     platforms=[
         'Operating System :: Microsoft :: Windows',
         'Operating System :: POSIX',
         'Operating System :: Unix',
         'Operating System :: MacOS',
     ],
-    classifiers=[
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8',
-        'Programming Language :: Python :: 3.9',
-        'Programming Language :: Python :: 3.10',
-        'Topic :: Software Development',
-        'Topic :: Scientific/Engineering',
-    ],
     packages=['jpype', 'jpype._pyinstaller'],
     package_dir={'jpype': 'jpype', },
     package_data={'jpype': ['*.pyi']},
-    install_requires=['typing_extensions ; python_version< "3.8"',
-        'packaging'],
-    tests_require=['pytest'],
-    extras_require={
-        'tests': [
-            'pytest',
-        ],
-        'docs': [
-            'readthedocs-sphinx-ext',
-            'sphinx',
-            'sphinx-rtd-theme',
-        ],
-    },
     cmdclass={
         'build_ext': setupext.build_ext.BuildExtCommand,
+        'develop': setupext.develop.Develop,
         'test_java': setupext.test_java.TestJavaCommand,
         'sdist': setupext.sdist.BuildSourceDistribution,
         'test': setupext.pytester.PyTest,
     },
     zip_safe=False,
     ext_modules=[jpypeJar, jpypeLib, ],
-    distclass=setupext.dist.Distribution,
```

Comment:
is the removed distclass the default value assumed in pyproject.toml?
Does anyone recall what this was needed for?

Reply:
Thanks for pointing this out. ``setupext.dist.Distribution`` on master enables the options (https://github.com/jpype-project/jpype/blob/master/setupext/dist.py#L25) ``enable-build-jar``, ``enable-tracing`` and ``enable-coverage``. In this PR, those options have become command options, and are duplicated in the ``setupext.develop.Develop`` and ``setupext.build_ext.BuildExtCommand`` commands. One of the reasons for doing this (if I remember correctly) is that distribution level options are not supported in the PEP-517 build back-end spec (I am dragging this up from deep in my memory, and https://github.com/pypa/pip/issues/5771 may be more insightful).

This change from global to command options is the reason for changes in the docs as:

```
python setup.py --enable-tracing develop
```

Becoming

```
python setup.py develop --enable-tracing
```

Reply:
Thanks for pointing this out. It is clear to me now.